frontend/src/routes/Credentials/CredentialsForm.tsx

```diff
@@ -936,15 +936,19 @@ export function CredentialsForm(props: {
         ],
         submit: () => {
             if (isEditing) {
-                return replaceResource(stateToData() as IResource).promise.then(async () => {
-                    if (process.env.NODE_ENV === 'development')
-                        await new Promise((resolve) => setTimeout(resolve, 4000))
+                const secret = stateToData() as Secret
+                const patch: { op: 'replace'; path: string; value: unknown }[] = []
```

Comment:
@jamestalton I think we need a little more code, there is a key in Spec.Data.`credential-hash` we need to include that from the current Spec.Data to the new Data set. Then update or patch works.

I think this flow will still remove the `credential-hash` because you overwrite (with patch) the whole data field.
Am I missing where the secret Spec.Data is aggregated?

Reply:
we will have to handle `credential-hash`